### PR TITLE
bwa: add arm64 support

### PR DESCRIPTION
**Pull request by alexreg**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/bwa.rb [hunk 1]
```diff
@@ -16,9 +17,13 @@ class Bwa < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux: "3d875f0b64143cfb58e44681e47102158b46e0004c408d8fc5f10dc976383a94"
   end
 
+  depends_on "sse2neon" => :build if Hardware::CPU.arch == :arm64
+
   uses_from_macos "zlib"
 
   def install
+    inreplace "ksw.c", "<emmintrin.h>", "<sse2neon.h>" if Hardware::CPU.arch == :arm64
```
Comment (chenrui333) [inline Formula/bwa.rb:25]: is there any upstream patch for this?

Comment (alexreg) [reply Formula/bwa.rb:25]: Nope, as I think you've just realised. (Thanks for making a note though.) The project is not very actively maintained, although it's mature. Maybe we can make a note of that PR here, merge as-is, and then remove the `inreplace` once/if the bug-fix gets accepted and a new release made?

Comment (chenrui333) [reply Formula/bwa.rb:25]: yeah, if you can submit an upstream PR and add a code comment reference that, that would be great.

Comment (cho-m) [reply Formula/bwa.rb:25]: There are 2 different upstream PRs, though both bundle corresponding library:
- `simde` - lh3/bwa#283
- `sse2neon` - lh3/bwa#344

Comment (alexreg) [reply Formula/bwa.rb:25]: @cho-m Yep. The first one is out-of-date and didn't work for me. The second is very simple and works fine. (Basically what I do with the `inreplace`.)

File: Formula/bwa.rb [hunk 2]
```diff
@@ -16,9 +17,14 @@ class Bwa < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux: "3d875f0b64143cfb58e44681e47102158b46e0004c408d8fc5f10dc976383a94"
   end
 
+  depends_on "sse2neon" => :build if Hardware::CPU.arch == :arm64
```
Comment (cho-m) [inline Formula/bwa.rb:20]:
```suggestion
  depends_on "sse2neon" => :build if Hardware::CPU.arm?
```

**Comment by alexreg** [comment]
@cho-m Resolved. Should be read for merge now.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**